File: app/src/test/java/com/example/mike/mp3player/dagger/modules/MockMediaControllerAdapterModule.java
package com.example.mike.mp3player.dagger.modules;

import android.content.Context;

import com.example.mike.mp3player.client.MediaControllerAdapter;
import com.example.mike.mp3player.client.MockMediaControllerAdapter;
import com.example.mike.mp3player.client.callbacks.MyMediaControllerCallback;

import dagger.Module;
import dagger.Provides;

@Module
public class MockMediaControllerAdapterModule  {


    @Provides
    MediaControllerAdapter provideMediaControllerAdapter(Context context, MyMediaControllerCallback myMediaControllerCallback) {
        return new MockMediaControllerAdapter(context, myMediaControllerCallback);
    }
}

File: android/client/src/unitTest/java/com/github/goldy1992/mp3player/client/dagger/modules/MockMediaControllerAdapterModule.kt
package com.github.goldy1992.mp3player.client.dagger.modules

import android.content.Context
import android.support.v4.media.MediaBrowserCompat
import com.github.goldy1992.mp3player.client.MediaControllerAdapter
import com.github.goldy1992.mp3player.client.callbacks.MyMediaControllerCallback
import com.github.goldy1992.mp3player.client.MockMediaControllerAdapter
import dagger.Module
import dagger.Provides

@Module
class MockMediaControllerAdapterModule {
    @Provides
    fun provideMediaControllerAdapter(context: Context?, mediaBrowserCompat: MediaBrowserCompat, myMediaControllerCallback: MyMediaControllerCallback): MediaControllerAdapter {
        return MockMediaControllerAdapter(context, mediaBrowserCompat, myMediaControllerCallback)
    }
}